### PR TITLE
Tab & Accordion: Add setting to disable scroll

### PR DESCRIPTION
**Pull request by AlexGStapleton**
Resolve https://github.com/siteorigin/so-widgets-bundle/issues/1163

### CODE REVIEW

File: widgets/accordion/accordion.php [hunk 1]
```diff
@@ -37,8 +37,30 @@ function initialize() {
...
+
+		add_action( 'siteorigin_widgets_enqueue_frontend_scripts_sow-accordion', array( $this, 'enqueue_widget_scripts' ) );
 	}
-	
+
+	function get_settings_form() {
+		return array(
+			'scrollto_after_change' => array(
+				'type'        => 'checkbox',
+				'label'       => __( 'Scroll user to top of the panel when opening panel', 'so-widgets-bundle' ),
```
Comment (Misplon) [inline widgets/accordion/accordion.php:48]: What do you think about renaming this label to:
"Scroll to top of the panel when opening panel"

File: widgets/accordion/accordion.php [hunk 2]
```diff
@@ -37,8 +37,30 @@ function initialize() {
...
+				'default'     => true,
+				'description' => __( 'This setting will scroll the user to the top of any opened panels. This is done to potentially prevent the need for the user to do this ', 'so-widgets-bundle' ),
```
Comment (Misplon) [inline widgets/accordion/accordion.php:50]: When opening the panel, scroll the user to the top of the panel.

**Comment by Misplon** [comment]
If you're ok with the above changes, perhaps let's make them for both widgets.

I'm getting some unusual results testing locally. Can you perhaps send me a build and I'll test it on the demo server?

**Comment by AlexGStapleton** [comment]
I've adjusted the setting name and descriptions. I also resolved a PHP compatibility issue and that may have been the error you were getting locally. Here's a build:

[so-widgets-bundle.scroll-top-settings.zip](https://github.com/siteorigin/so-widgets-bundle/files/5322398/so-widgets-bundle.scroll-top-settings.zip)

**Comment by Misplon** [comment]
Thanks. I've uploaded this build to the demo server: https://demo.siteorigin.com/corp/accordion/. Both locally and on the demo server, the scroll function doesn't seem to be functioning.

**Comment by AlexGStapleton** [comment]
It can be sometimes quite tricky to trigger the scroll with the Accordion widget. In Corp, the easiest way would be to disable the Sticky Header or remove the header altogether using the inspector. Then scroll to the point where the accordion panel heading is barely visible and then click it. I considered suggesting not implementing this change for the Accordion widget but I received a report from a user and their theme seemed to result in this happening on a fairly consistent basis regardless of where the viewport where positioned.

Here's a video:

[](https://drive.google.com/uc?id=1e0B63ChfVyVoowGM2nSeaFa2rHxa9Mfm)

The easiest way to replicate this with the tabs widget is to change the tab placement.

**Comment by Misplon** [comment]
Thanks. Seems fine for Accordion but for me, scroll top seems to be always off for the Tabs widget. I put the tabs at the bottom of the container with a lot of text in each tab and I don't seem to be scrolled up. Do you?

**Comment by Misplon** [comment]
I think for me, the Tabs setting is working in reverse. If I deactivate the Scroll top setting then I scroll to the top.

**Review by Misplon — APPROVED** [review]

**Review by gregpriday — APPROVED** [review]

**Pull request merged**